### ISSUE TITLE
ZFS snapshots are a rabbit hole

### ISSUE THREAD

**Issue by bachmarc** [nextcloud/previewgenerator#486]
HI,
I tried that app and relaized it tries to permutate through all folders.

The folder "dot"  .zfs is a default virtual folder that contains all zfs snapshots of the past which will show my file system in read only mode as it was the day before and so on... 
That means 
a) I cannot add a .nomedia file
b) it contains all my images over and over and over again with a different path as my server creates a sbapshot every day....

and it failed anyway ;) but I won't be able to debug... 

2024-06-25T13:01:42+00:00 Scanning folder /main/files
2024-06-25T13:01:42+00:00 Scanning folder /main/files/content_main
2024-06-25T13:01:42+00:00 Scanning folder /main/files/content_main/.Trash-1000
2024-06-25T13:01:42+00:00 Scanning folder /main/files/content_main/.Trash-1000/files
2024-06-25T13:01:42+00:00 Scanning folder /main/files/content_main/.Trash-1000/files/bridge
2024-06-25T13:01:42+00:00 Scanning folder /main/files/content_main/.Trash-1000/info
2024-06-25T13:01:42+00:00 Scanning folder /main/files/content_main/.zfs
2024-06-25T13:01:42+00:00
2024-06-25T13:01:42+00:00 In Parser.php line 157:
2024-06-25T13:01:42+00:00
2024-06-25T13:01:42+00:00   Malformed state response from server
2024-06-25T13:01:42+00:00

**Comment by st3iny** [comment]
I assume you are talking about the initial generation. You can limit the initial generation to certain folders by using the `--path` parameter of `preview:generate-all`.

The incremental generation should not pick up changes inside `.zfs` as they are not changed from within Nextcloud.

**Comment by bachmarc** [comment]
Yes, that is what I did afterwards ... but only because its a small installation. Maybe you filter out .zfs in default as it always contains redundant data that is not live. With 200 users admins may struggle

THX Marc

**Comment by st3iny** [comment]
Filtering by default is not something I'm considering. However, adding an exclude option to the initial generation command would be a nice feature.
